src/controllers/v1/_common/demand.ts

```diff
@@ -12,13 +12,15 @@ import {
 } from '../../../models/demand'
 import { DATE, UUID } from '../../../models/strings'
 import { BadRequest, NotFound } from '../../../lib/error-handler/index'
-import { getMemberByAddress, getMemberBySelf } from '../../../lib/services/identity'
+import IdentityClass from '../../../lib/services/identity'
 import { TransactionResponse, TransactionType } from '../../../models/transaction'
 import { demandCommentCreate, demandCreate } from '../../../lib/payload'
 import ChainNode from '../../../lib/chainNode'
 import env from '../../../env'
 import { parseDateParam } from '../../../lib/utils/queryParams'
 
+const identityClass = new IdentityClass()
```

Comment:
We shouldn't be building this here. Ideally we should be instantiating the instance of IdentityClass (I dislike this name btw but it's ok) and passing this into the constructor of the controller. Look at https://github.com/digicatapult/dscp-matchmaker-api/blob/main/src/controllers/health/index.ts and https://github.com/digicatapult/dscp-matchmaker-api/blob/main/src/lib/service-watcher/index.ts for an example. I'm not sure exactly how parameters are passed to the singleton constructor; it might be best to look at documentation in https://github.com/microsoft/tsyringe and https://tsoa-community.github.io/docs/di.html#tsyringe